api/server/server_options.go

```diff
@@ -10,6 +10,8 @@ import (
 
 type ServerOption func(context.Context, *Server) error
 
+type RIDHandler func(string) func(c *gin.Context)
```

Comment:
Thinking out loud here:

Would it make sense to have stronger semantics around RID? 

We know it is header (and context value), we want to set it if it is not set, we want some string value provider if the header is not set, we want the header parameter name to be configurable

What about having something like:

`type RIDProvider struct {
	Name string
	ProvideNew func() string
}`

and then:

`WithRIDProvider(ridProvider RIDProvider) ServerOptions`

This way we would not need to leak the gin.Context and also the 'if the value not set, set new value' logic might be handled by fn server.